packages/property-view/src/browser/resource-property-view/resource-property-data-service.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import URI from '@theia/core/lib/common/uri';
+import { PropertyDataService } from '../property-data-service';
+import { FileSelection } from '@theia/filesystem/lib/browser/file-selection';
+import { Navigatable } from '@theia/core/lib/browser/navigatable';
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+disableJSDOM();
+
+let resourcePropertyDataService: ResourcePropertyDataService;
+
+const MockFileStat: FileStat = {
+    isFile: false,
+    isDirectory: true,
+    isSymbolicLink: false,
+    resource: new URI('resource'),
+    name: 'name'
+};
+
+describe('resource-property-data-service', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertyDataService).toSelf().inSingletonScope();
+        container.bind(FileService).toConstantValue({
+            async resolve(_uri: URI): Promise<FileStat> {
+                MockFileStat.resource = _uri;
+                return MockFileStat;
+            }
+        } as FileService);
+        container.bind(PropertyDataService).to(ResourcePropertyDataService).inSingletonScope();
+        resourcePropertyDataService = container.get(ResourcePropertyDataService);
+    });
+
+    after(() => {
+        disableJSDOM();
+    });
+
+    const fileSelection: FileSelection = {
+        fileStat: MockFileStat
+    };
+
+    const navigatableSelection: Navigatable = {
+        getResourceUri(): URI | undefined {
+            const uri = new URI('uri');
+            return uri;
+        },
+        createMoveToUri(): URI | undefined {
+            const newUri = new URI('newURI');
+            return newUri;
+        }
+    };
+
+    const arrayFileSelection: FileSelection[] =
+        [fileSelection, { fileStat: { ...MockFileStat, resource: new URI('secondURI') } }];
+
+    describe('#canHandle', () => {
+
+        it('should handle an empty object', () => {
+            expect(resourcePropertyDataService.canHandleSelection({})).eq(0);
+        });
+
+        it('should handle undefined', () => {
+            expect(resourcePropertyDataService.canHandleSelection(undefined)).eq(0);
+        });
+
+        it('should handle a file selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(arrayFileSelection)).to.be.greaterThan(0);
+        });
+
+        it('should handle a navigatable selection', () => {
+            expect(resourcePropertyDataService.canHandleSelection(navigatableSelection)).to.be.greaterThan(0);
+        });
+    });
+
+    describe('#providePropertyData', () => {
+
+        it('should return the stat of a fileSelection', () => {
+            const promise = resourcePropertyDataService.providePropertyData(arrayFileSelection);
+            promise
+                .then(fileStat => expect(fileStat).eq(MockFileStat))
+                .catch(error => console.log(error));
+        });
```

Comment:
You can make it `async` and therefore do not need to `.then`:

```suggestion
        it('should return the stat of a fileSelection',  async () => {
            const data = await resourcePropertyDataService.providePropertyData(arrayFileSelection);
            expect(data).to.equal(mockFileStat);
        });
```